pingrid/impl.py

```diff
@@ -96,24 +87,40 @@ class ColorScale:
     
     def __init__(self, colors, scale=None):
         self.colors = colors
-        self.scale = scale
-    
-    def cv2cm(self):
-        if self.scale is None:
-           cs_val = np.arange(len(colors))
+        if scale is None:
+            self.scale = list(np.arange(len(colors)))
         else:
-           if len(self.colors) == len(self.scale):
-               cs_val = np.array(self.scale)
+           if len(colors) == len(scale):
+               self.scale = scale
            else:
-               raise Exception("if provided, scale must be same length as colors")
-        cv2cm_i = (255*(cs_val-cs_val[0])/(cs_val[-1]-cs_val[0])).astype(int)
-        cv2cm_i = cv2cm_i + np.append([0], np.where(np.diff(cv2cm_i) == 0, 1, 0))
-        cv2cm_rgba = np.array(self.colors)
-        cv2cm_bgra = cv2cm_rgba[:,[2, 1, 0, 3]]
-        cv2cm = np.full((256,4), np.nan)
-        for bgra in range(4):
-            cv2cm[:, bgra] = np.interp(np.arange(256), cv2cm_i, cv2cm_bgra[:, bgra])
-        return cv2cm.astype(int)
+               raise Exception(
+                   "if provided, scale must be same length as colors"
+               )
+    
+    def flip_colors(self):
+        return ColorScale(self.colors[::-1], self.scale)
+
+    def to_rgba(self):
+        cs_val = np.array(self.scale)
+        cs_i = (255*(cs_val-cs_val[0])/(cs_val[-1]-cs_val[0])).astype(int)
+        cs_i = cs_i + np.append([0], np.where(np.diff(cs_i) == 0, 1, 0))
+        cs_rgba = np.array(self.colors)
+        cs_rgba_full = np.full((256,4), np.nan)
+        for rgba in range(4):
+            cs_rgba_full[:, rgba] = np.interp(np.arange(256), cs_i, cs_rgba[:, rgba])
+        return cs_rgba_full.astype(int)
+
+    def to_bgra(self):
+        return self.to_rgba()[:,[2, 1, 0, 3]]
+
+    def to_hex(self):
+        cm = self.to_rgba()
+        cs = []
+        for x in cm:
+            v = RGBA(*x)
+            cs.append(f"#{v.red:02x}{v.green:02x}{v.blue:02x}{v.alpha:02x}")
```

Comment:
We could have a `to_hex` method  on the `RGBA` class, and then call that method here.

The same might be true for the part of `to_rgba` that calculates the representation for a single color.

Reply:
I did something along those lines